docs/MODULE_INTERFACE.md

```diff
@@ -0,0 +1,57 @@
+# Station module interface
+
+1. A module itself is a long-running executable, which
+    - performs its business logic
+    - writes informative messages to `stdio`
+    - exposes stats via a builtin webserver
+
+1. A module needs to have its __binaries hosted__ on HTTPS accessible URLs, so that Station can download updates. Path patterns are flexible, GitHub Releases would be nice.
+
+    Example:
+    ```
+    https://github.com/filecoin-saturn/L2-node/releases
+    ```
+
+1. A module should run on as many operating systems as possible, without requiring any prior setup before executing its binary.
+
+2. A module can provide a custom __command string__ for launching its executable. Ideally a module doesn't require any extra arguments when launching.
+
+    Example:
+    ```bash
+    $ bacalhau # ideal
+    $ bacalhau --foo=bar
+    ```
+
+3. A module's executable gets passed the following environment variables:
+    - `FIL_WALLET_ADDRESS` The user's Filecoin wallet address
+    - `ROOT_DIR` The long-lived working directory on disk. The module must store
+      all of its files inside (subdirectories of) this directory. The directory
+      isn't expected to be backed up or shared across machines in any way.
+
+    Example:
+    ```bash
+    $ FIL_WALLET_ADDRESS=f1... ROOT_DIR=/var/bacalhau bacalhau
+    ```
+
+4. A module communicates activity by writing to its `stdout` stream:
+    - `API: $1` The module has launched and `$1` can be queried for fetching module stats (see below)
+    - `INFO: $1` `$1` will be displayed to the user inside Station's Activity Log
+    - `ERROR: $1` `$1` will be displayed to the user inside Station's Activity Log, with a warning icon
+    - `$1` all log lines will be stored in a module-specific log file, which can be submitted to Sentry for error handling
+
+5. A module is expected to have provided its `API` URL in `<=500ms`
+
+6. A module's `stderr` will be stored in the same module-specific log file as its `stdout`, to be used for post-mortem debugging.
+
+7. A module exposes its stats via HTTP(S), on an address communicated via `stdout` (see above `API:`). The response will be a JSON object. It is for Station to interpret the response and update its UI accordingly. The responses should be as consistent as possible.
```

Comment:
What are the APIs the module needs to expose?

Reply:
We just added some clarifications to this section, is this sufficient now?